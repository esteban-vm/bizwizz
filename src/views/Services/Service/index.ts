export { default, type ServiceProps } from './Service'
