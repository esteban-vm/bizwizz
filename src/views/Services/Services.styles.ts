import clsx from 'clsx'

const styles = {
  services: clsx('m-[0_6rem] flex flex-wrap gap-6'),
}

export default styles
