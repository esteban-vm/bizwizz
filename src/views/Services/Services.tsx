import type { FC } from '@/types'
import { Wave } from '@/components'
import Service, { type ServiceProps } from './Service'
import styles from './Services.styles'

export const services: ServiceProps[] = [
  { id: '92cf33d6-5464-4bb5-8768-d62f81ab5036', title: 'Digital Marketing', icon: 'strategy' },
  { id: '3d87ca25-fe39-47f3-8d1c-3cdf0f459a94', title: 'UX Design', icon: 'desktop' },
  { id: 'fdbe44b3-5dca-4f76-b07a-e6de8b0bf480', title: 'Product Design', icon: 'picture' },
  { id: '4706f6ad-30b1-4eca-a8a3-5571508a9949', title: 'Content Writing', icon: 'pencil' },
  { id: '0d3834cc-d18c-49db-97b0-e0c06aefc6fc', title: 'Concept Development', icon: 'expand' },
  { id: '8718afff-4203-47cb-a1f1-4232466284d4', title: 'Social Outreach', icon: 'megaphone' },
  { id: '1e23a19c-4a83-4ab8-a05d-adefccfae2f1', title: 'Media Planning', icon: 'video' },
  { id: '6ef025ef-fd35-474e-9090-0f9e19ff2ef4', title: 'Branding', icon: 'trophy' },
  { id: '47bf22db-4831-47bd-8d97-fc2bb53471b3', title: 'SEO', icon: 'map' },
]

const Services: FC = () => {
  return (
    <section aria-labelledby='services_title' className='section bg-white' id='services'>
      <Wave is='black' />
      <div className='section_header'>
        <h2 className='section_heading' id='services_title'>
          Our services
        </h2>
        <h3 className='section_subheading'>Lorem ipsum dolor sit amet</h3>
      </div>
      <div className={styles.services}>
        {services.map((service) => (
          <Service key={service.id} {...service} />
        ))}
      </div>
    </section>
  )
}

export default Services
