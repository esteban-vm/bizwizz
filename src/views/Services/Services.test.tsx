import { render, cleanup, within, screen } from '@/tests'
import Services from './Services'

describe('Services:', () => {
  let asFragment: () => DocumentFragment

  beforeEach(() => {
    void ({ asFragment } = render(<Services />))
  })

  afterEach(cleanup)

  it('should match snapshot', () => {
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render correctly', () => {
    const services = screen.getByRole('region', { name: /our services/i })
    expect(services).toBeInTheDocument()
    expect(within(services).getByRole('heading', { level: 2 })).toBeInTheDocument()
    expect(within(services).getByRole('heading', { name: /lorem ipsum dolor sit amet/i, level: 3 })).toBeInTheDocument()
  })
})
