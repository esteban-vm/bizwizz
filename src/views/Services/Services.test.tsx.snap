// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Services: > should match snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="services_heading"
    class="bg-white pb-32 text-center"
    id="services"
  >
    <svg
      class="text-black"
      height="10rem"
      preserveAspectRatio="none"
      role="presentation"
      viewBox="0 0 500 150"
      width="100%"
    >
      <path
        d="M0.00,49.98 C161.96,151.48 356.65,-54.75 500.00,49.98 L504.51,-36.00 L-13.54,-3.44 Z"
        fill="currentColor"
        stroke="none"
      />
    </svg>
    <div
      class="font-bold tracking-widest"
    >
      <h2
        class="mb-5 text-[2.4rem] uppercase text-primary"
        id="services_heading"
      >
        Our services
      </h2>
      <h3
        class="mb-10 text-[4rem] capitalize text-darkgrey"
      >
        Lorem ipsum dolor sit amet
      </h3>
    </div>
    <div
      class="m-[0_6rem] flex flex-wrap gap-6"
    >
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="92cf33d6-5464-4bb5-8768-d62f81ab5036"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-strategy"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Digital Marketing
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="3d87ca25-fe39-47f3-8d1c-3cdf0f459a94"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-desktop"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          UX Design
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="fdbe44b3-5dca-4f76-b07a-e6de8b0bf480"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-picture"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Product Design
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="4706f6ad-30b1-4eca-a8a3-5571508a9949"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-pencil"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Content Writing
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="0d3834cc-d18c-49db-97b0-e0c06aefc6fc"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-expand"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Concept Development
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="8718afff-4203-47cb-a1f1-4232466284d4"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-megaphone"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Social Outreach
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="1e23a19c-4a83-4ab8-a05d-adefccfae2f1"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-video"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Media Planning
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="6ef025ef-fd35-474e-9090-0f9e19ff2ef4"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-trophy"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          Branding
        </h4>
      </article>
      <article
        class="
        flex
        w-80
        flex-[1_1_30%]
        items-center
        justify-center
        overflow-hidden
        bg-lightgrey
        text-black
        hover:bg-primary
        hover:text-white
        [&>h4]:hover:translate-x-20
        [&>svg]:hover:scale-150
        [&>svg]:hover:opacity-60
      "
        id="47bf22db-4831-47bd-8d97-fc2bb53471b3"
      >
        <svg
          class="
          mr-7
          h-44
          w-[4.8rem]
          scale-50
          fill-primary-light
          opacity-0
          transition-all
          duration-500
        "
          role="img"
        >
          <use
            xlink:href="#icon-map"
          />
        </svg>
        <h4
          class="
          select-none
          p-[3rem_6rem_3rem_0]
          text-[2.3rem]
          font-[500]
          transition-all
          duration-500
        "
        >
          SEO
        </h4>
      </article>
    </div>
  </section>
</DocumentFragment>
`;
